plugin/pkg/scheduler/algorithm/scheduler_interface.go

```diff
@@ -34,6 +34,11 @@ type SchedulerExtender interface {
 	// are used to compute the weighted score for an extender. The weighted scores are added to
 	// the scores computed  by Kubernetes scheduler. The total scores are used to do the host selection.
 	Prioritize(pod *v1.Pod, nodes []*v1.Node) (hostPriorities *schedulerapi.HostPriorityList, weight int, err error)
+
+	// Bind informs the extender about the scheduling decision, before the decision is
+	// conveyed to the apiserver. The binding information contains the pod and the
+	// selected node.
+	Bind(pod *v1.Pod, node string) error
```

Comment:
Here's the case in my mind:

1. S->E: Filter request: n1, n2, n3, n4
1. E->S: n1, n2, n3
1. S->E: prioritize request: n1, n2, n3
1. E->S: n1, n2, n3
1. E: n1 is not valid, e.g. revoked by Mesos
1. S->E: bind Pod to n1
1. E->S: reject

If we changed step 6 to `S-E: bind Pod to candidates: n1, n2, n3`, extender will return successful at step 7, e.g. `E-S: bind to n2`. But the problem is that: the assumed pods are also considered for affinity/anti-affnity; if we assumed all candidates (e.g. n1, n2, n3), it'll make scheduler slower.

As a conclusion, I'm OK to keep current interface to send only one host: step 5 is less case compared to normal task launching :).